node/compile-hook.js

```diff
@@ -11,10 +12,31 @@ if (! _compile.reified) {
   (Mp._compile = function (content, filename) {
     return _compile.call(
       this,
-      // Don't touch the file unless reification is enabled for the
-      // package that contains the file.
-      compile(content, filename),
+      compile(content, getOptions(filename)),
       filename
     );
   }).reified = _compile;
 }
+
+function getOptions(filename) {
+  const options = { filename: filename };
+  const mtime = mtimeOrNull(filename);
+
+  if (mtime !== null) {
+    options.cacheKey = {
+      source: "Module.prototype._compile",
+      filename: filename,
+      mtime: mtime
+    };
+  }
+
+  return options;
+}
+
```

Comment:
I think there's already a `statOrNull` helper somewhere maybe a `node/util.js` would be handy.